plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -264,6 +261,25 @@ public ResultSet executeQuery()
         };
     }
 
+    public PreparedStatement prepareStatement(
+            ConnectorSession session,
+            Connection connection,
+            JdbcTableHandle table,
+            List<JdbcColumnHandle> columns,
+            Optional<JdbcSplit> split)
+            throws SQLException
+    {
+        PreparedQuery preparedQuery = prepareQuery(
+                session,
+                connection,
+                table,
+                Optional.empty(),
+                columns,
+                ImmutableMap.of(),
+                split);
+        return new QueryBuilder(this).prepareStatement(session, connection, preparedQuery);
```

Comment:
Does this change mean we can return true from `isLimitGuaranteed` in Phoenix?

Reply:
No. If you pass Phoenix a limit > Integer.MAX_VALUE it will still completely ignore it, so `isLimitGuaranteed` has to return false.

Reply:
but that's JdbcMetadata responsibility, not PhoenixClient, right? PhoenixClient doesn't ignore anything.